### PR TITLE
Allow for perspective in ipywidgets layouts

### LINKED ISSUE #694: Hook perspective into ipywidgets
`if (this.pWidget) { this.pWidget.add(this.psp); }`
@vidartf

### PR DESCRIPTION
**Pull request by timkpaine**
fixes #694 without breaking in normal view

top: perspective in `ipywidgets.HBox`
bottom: normal

Screen Shot 2019-09-02 at 2 54 40 PM

### CODE REVIEW

**Comment by timkpaine** [comment]
@vidartf

File: packages/perspective-jupyterlab/src/ts/widget.ts [hunk 1]
```diff
@@ -98,7 +96,21 @@ class PerspectiveView extends DOMWidgetView {
...
              wrap: false, // wrap: handled by perspective-python
              delete_: true, // delete_: handled by perspective-python
+             view: this, // necessary for ipywidgets
         });
+        return this.pWidget.node;
+    }
+
+    _setElement(el: HTMLElement) {
+        if (this.el || el !== this.pWidget.node) {
+            // Accordions don't allow setting the element beyond the initial creation.
```
Comment (vidartf) [inline packages/perspective-jupyterlab/src/ts/widget.ts]: Probably shouldn't say "Accordions" ;)

File: packages/perspective-jupyterlab/src/ts/widget.ts [hunk 2]
```diff
@@ -119,98 +131,98 @@ class PerspectiveView extends DOMWidgetView {
         this.model.on('msg:custom', this._update, this);
 
         this.displayed.then(()=> {
-            this.psp._render();
+            (this.pWidget as PerspectiveWidget)._render();
```
Comment (vidartf) [inline packages/perspective-jupyterlab/src/ts/widget.ts]: Would be cleaner to override the declaration of `pWidget` for this class:

```ts
pWidget: PerspectiveWidget;
```

File: packages/perspective-phosphor/src/ts/widget.ts [hunk 3]
```diff
@@ -44,6 +44,7 @@ export type PerspectiveWidgetOptions = {
     key?: string;
     wrap?: boolean;
     delete_?: boolean;
+    view?: any;
```
Comment (vidartf) [inline packages/perspective-phosphor/src/ts/widget.ts]: Note that it is also possible to `extend` `PerspectiveWidget` adding the `view` options similar to how `JupyterPhosphorAccordionWidget` does this in `@jupyter-widgets/controls`.

**Pull request merged**